h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java

```diff
@@ -169,7 +169,7 @@
       if (! hasGPU(_parms._gpu_id))
         error("_backend", "GPU backend (gpu_id: " + _parms._gpu_id + ") is not functional. Check CUDA_PATH and/or GPU installation.");
 
-      if (H2O.getCloudSize() > 1 && !_parms._build_tree_one_node)
+      if (H2O.getCloudSize() > 1 && !_parms._build_tree_one_node && !allowMultiGPU())
```

Comment:
Shouldn't the below error message updated as well?

Reply:
@mn-mikke I wouldn't think so - if multigpu is enabled then no one will see this error. Or?

Reply:
This just really minor comment I won't be against if you leave it as it is.

Just If I saw this message, i would understand that there is no way how to enable it for distributed mode, but it will be possible with `allowMultiGPU` option, won't it?

Reply:
I think I see where the misunderstanding is -> multi gpu currently requires distributed (multiple nodes even on a single machine). So the error will never be shown until it is really distributed.

Reply:
Do you mean by 'distributed' the option `_parms._build_tree_one_node`?

Reply:
No, for multi-gpu to work (even on a single node) you will need 2+ instances of H2O.

So the error will only be shown if someone has 2+ instances and `allowMultiGPU = false` (which is the default), all assuming the is regular model build not single-node tree building.

I am starting to think that you want to modify the error message to include `_parms._build_tree_one_node`? Is that so? Because just for the change I made - I don't see how I would change the message :)

Maybe if you could propose a new error message - it would be clearer to me.

Reply:
The message `GPU backend is not supported in distributed mode.` sounds to me that GPU is not supported for distrtibuted method at all.

I have been wondering if we change it to something like: `GPU backend for the distributed mode is currently disabled. It can be enabled by setting the property 'xgboost.multinode.gpu.enabled' to true.'`

Anyway if you think that the original message is sufficient, I'm happy with that. I don't want to bother you with such a minor thing. I assume there are more important things to solve.

Reply:
since `xgboost.multinode.gpu.enabled` is experimental I dont think we want to mention it in our error messages and encourage people using it.

Reply:
@mn-mikke got it, this is now meant for internal experimentation and for controlled public experimentation but generally, users shouldn't be turning it on

so I think keeping the message as-is makes sense, thanks for the comment and the discussion!

Reply:
> this is now meant for internal experimentation and for controlled public experimentation but generally, users shouldn't be turning it on

ok, no problem. makes sense